src/sign.ml

```diff
@@ -277,6 +282,29 @@ let add_rule : t -> sym -> rule -> unit = fun sign sym r ->
     let m = (sym.sym_name, r) :: m in
     sign.sign_deps := PathMap.add sym.sym_path m !(sign.sign_deps)
 
+(** [add_rules s r] adds a bunch of rules [r] to its attached symbol to the
+    signature [s] and builds decision trees. *)
+let add_rules : t -> (sym * pp_hint * rule loc) list -> unit = fun sign rs ->
+  let add_rule (s, h, r) =
+    out 3 "(rule) %a\n" Print.pp_rule (s,h,r.elt) ;
+    add_rule sign s r.elt in
+  List.iter add_rule rs ;
+  (* EXPE *)
+  let build_tree symb =
+    match symb.sym_mode with
+    | Defin
+    | Injec when !(symb.sym_rules) <> [] ->
+       let pama = lazy (Dtree.ClauseMat.of_rules !(symb.sym_rules)) in
+       let tree = lazy (Dtree.compile @@ Lazy.force pama) in
```

Comment:
What is @@? Where is it defined?

Reply:
`@@` is the application.  It avoids enclosing in parentheses sometimes.  Regarding whether it is better to enclose in parenthesis or use `@@`, I do not have a clear idea.  It is like the `$` in haskell.

Reply:
I didn't know this notation. @rlepigre what do you think: for the sake of uniformity, do we keep this notation?

Reply:
I tend to avoid Haskell-like notations. For me, this is nothing but code obfuscation, or another way to make code impossible to understand to "regular" programmers. And the same goes for monadic notations.

Reply:
Ah, the `|>` appears quite handy when chaining various maps or filter (and is, on the overall, more justified than the `@@` imo).

Reply:
Yeah, but it is easy to forget that chaining several maps / filters is usually a bad idea for efficiency.

Reply:
I see the point